### PR TITLE
Migration will now pick win7-x86 as the RuntimeIdentifier to be used by Full Framework TFMs

### PR DESCRIPTION
**Pull request by livarcocc**
Migration will now pick win7-x86 as the RuntimeIdentifier to be used by Full Framework TFMs when no RIDs exist in the source PJ. This will allow build/publish etc to continue working in csproj without extra RIDs parameters in the command line.

Fixes https://github.com/dotnet/cli/pull/4991.
Fixes https://github.com/dotnet/cli/pull/5141.

@piotroko @piotrpMSFT @krwq @jgoshi @jonsequitur @eerhardt

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectJsonMigration/ProjectContextExtensions.cs [hunk 1]
```diff
@@ -17,5 +17,30 @@ public static string GetProjectName(this ProjectContext projectContext)
...
+        {
+            return projectContexts.Any(p => p.ProjectFile.Runtimes.Any());
+        }
+
+        public static bool HasBothCoreAndFullFrameworkTFMs(this IEnumerable<ProjectContext> projectContexts)
+        {
+            return projectContexts.HasCoreTFM() && projectContexts.HasFullFrameworkTFM();
+        }
+
+        public static bool HasCoreTFM(this IEnumerable<ProjectContext> projectContexts)
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.ProjectJsonMigration/ProjectContextExtensions.cs]: Can we expand the abbreviations?

File: src/Microsoft.DotNet.ProjectJsonMigration/transforms/AddPropertyTransform.cs [hunk 2]
```diff
@@ -32,7 +32,13 @@ public AddPropertyTransform(string propertyName, Func<T, string> propertyValueFu
 
         public AddPropertyTransform<T> WithMSBuildCondition(string condition)
         {
-            _msbuildCondition = condition;
+            _msbuildConditionFunc = source => condition;
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.ProjectJsonMigration/transforms/AddPropertyTransform.cs]: Is null or empty valid? This kind of indirection can be pretty hard to debug if we accept bad data to be used in a deferred call.

**Comment by livarcocc** [comment]
@srivatsn

**Comment by piotrpMSFT** [comment]
@srivatsn

**Pull request merged**